src/NHibernate/Util/LinkedHashMap.cs

```diff
@@ -385,7 +400,7 @@ public override string ToString()
 
 		#endregion
 
-		private class KeyCollection : ICollection<TKey>
+		public class KeyCollection : ICollection<TKey>
```

Comment:
To avoid increasing the API for which we have to avoid breaking changes, better switch it to internal instead, if possible.

Reply:
I cannot make it internal because it is used in a public method.
I would suggest to make `LinkedHashMap` internal.